core/BranchPred.cpp

```diff
@@ -0,0 +1,70 @@
+#include "BranchPred.hpp"
+
+/*
+ * The algorithm used for prediction / update is as follows:
+ * Prediction:
+ *    - look up BHT to determine if the branch is predicted taken or not
+ *    - look up BTB to see if an entry exists for the input fetch pc
+ *       - if present in BTB and predicted taken, BTB entry is used to determine
+ *         prediction branch idx and predictedPC
+ *       - if present in BTB but predicted not taken, BTB entry is used to determine
+ *         prediction branch idx, while predictedPC is the fall through addr
+ *       - if not present in BTB entry, prediction branch idx is the last instr of
+ *         the FetchPacket, while predicted PC is the fall through addr. Also, create
+ *         a new BTB entry 
+ * Update:
+ *    - a valid BTB entry must be present for fetch PC
+ *    - TBD 
+ *
+ */
+
+#define BYTES_PER_INST 4
+
+void SimpleBranchPredictor::updatePredictor(DefaultUpdate & update) {
```

Comment:
Does not appear that the `DefaultUpdate` object is modified in this function.  Make it `const`